### PR TITLE
feat(costoptimizationhub): report Savings Plans opportunities

### LINKED ISSUE #200: Report SageMaker Savings Plans recommendations
## What

Turn SageMaker Savings Plans recommendations from AWS Cost Optimization Hub into CloudBurn findings.

## Why

The account audit found one `PurchaseSavingsPlans` recommendation for SageMaker with estimated savings of $107.85 per month, or 26%. CloudBurn didn't report it.

AWS classifies SageMaker Savings Plans as a very-low-effort recommendation that doesn't require a restart:
https://docs.aws.amazon.com/cost-management/latest/userguide/coh-optimization-strategies.html

## Acceptance criteria

- Load SageMaker Savings Plans recommendations through the account-scoped Cost Optimization Hub dataset.
- Preserve estimated monthly cost, estimated savings, savings percentage, term, payment option, refresh time, and recommendation source.
- Report a diagnostic when Cost Optimization Hub is unavailable or access is denied.
- Deduplicate a recommendation if a native CloudBurn rule already reports the same action.
- Add discovery tests for recommendation, empty, paginated, and denied responses.

Focused implementation slice for #169.

### PR DESCRIPTION
**Pull request by dannysteenman**
## Summary

- Add `CLDBRN-AWS-COSTOPTIMIZATIONHUB-1` for Compute, EC2 Instance, and SageMaker Savings Plans purchase recommendations from AWS Cost Optimization Hub.
- Rework `CLDBRN-AWS-SAGEMAKER-3` into a coverage-gap check backed by the last 30 complete days of Cost Explorer data.
- Prevent duplicate SageMaker findings when Cost Optimization Hub already has a purchase recommendation for the same account.
- Preserve incomplete, denied, and unavailable evidence as `not_applicable` instead of reporting a clean pass.

## Why

Savings Plans purchase recommendations share one Cost Optimization Hub action and evidence model, so one generic rule gives them a consistent result shape. SageMaker coverage answers a different question: it catches material uncovered usage before AWS emits a purchase recommendation.

The coverage rule uses an 80 percent coverage threshold and requires at least 72 cost units of uncovered On-Demand usage. Cost Optimization Hub evidence is optional for that rule: it is used for deduplication only when another active rule already requested it, and never triggers extra Hub calls by itself.

## Scope

- [x] `cloudburn` CLI
- [x] `@cloudburn/sdk`
- [x] `@cloudburn/rules`
- [x] Documentation and changesets

## Verification

- [x] `pnpm verify`
- [x] `pnpm exec turbo boundaries`
- [x] `pnpm build`
- [x] Focused loader, discovery, scanner, and rule tests
- [x] Updated compiled CLI validated against the requested read-only AWS scope

Live validation confirmed recommendation reporting, materiality gating, same-account duplicate suppression, and explicit unavailable-data diagnostics. No customer or account-specific output is included here.

## Related issues

Closes #200
Related to #169

### CODE REVIEW

**Comment by dannysteenman** [comment]
> Re: comment 5540715199

No actionable feedback was included in this review-status summary.

File: packages/sdk/src/providers/aws/resources/cost-optimization-hub.ts [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+        () =>
+          client.send(
+            new ListRecommendationsCommand({
+              filter: {
+                accountIds: [accountId],
+                actionTypes: ['PurchaseSavingsPlans'],
+                resourceTypes: ['SageMakerSavingsPlans'],
+              },
+              includeAllRecommendations: false,
+              maxResults: PAGE_SIZE,
```
[the remark this answers is not in the document]
Comment (dannysteenman) [reply packages/sdk/src/providers/aws/resources/cost-optimization-hub.ts:184]: Won't fix: AWS documents a maximum value of 1000 for ListRecommendations maxResults, and the request has also been verified against the live API. Existing nextToken pagination remains in place. https://docs.aws.amazon.com/aws-cost-management/latest/APIReference/API_CostOptimizationHub_ListRecommendations.html

**Comment by dannysteenman** [comment]
> Re: comment 5540769486

No action needed: the combined review found no issues.

**Comment by dannysteenman** [comment]
> Re: comment 5540769486

Already addressed. This review reported no actionable findings.

**Comment by dannysteenman** [comment]
> Re: comment 5540715199

Already addressed. This is a PR-level review-status summary and does not contain an actionable finding.

File: packages/sdk/src/providers/aws/resources/cost-optimization-hub.ts [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+            message:
+              'Skipped Savings Plans recommendations because AWS Cost Optimization Hub returned incomplete recommendation evidence.',
+            provider: 'aws',
+            service: 'costoptimizationhub',
+            source: 'discovery',
+            status: 'skipped',
+          },
+        ],
+        resources: recommendations,
+        unavailable: true,
```
[the remark this answers is not in the document]
Comment (dannysteenman) [reply packages/sdk/src/providers/aws/resources/cost-optimization-hub.ts:221]: Fixed in a26ad72. Rules now evaluate with unavailable optional datasets masked from their resource bag, so partial Hub evidence cannot suppress the SageMaker fallback while COSTOPTIMIZATIONHUB-1 is not applicable. Added regression coverage for the mixed valid/incomplete recommendation case.

**Comment by dannysteenman** [comment]
> Re: comment 5540769486

Won’t fix: verified against the live Cost Explorer API on 2026-09-04. GetDimensionValues returns the SERVICE value Amazon SageMaker, and GetSavingsPlansCoverage with that exact filter returns populated coverage. The SageMaker AI product rename did not change this billing dimension, so changing the constant would break the rule.
